dracut/modules.d/90kiwi-live/module-setup.sh

```diff
@@ -29,7 +29,7 @@ install() {
     inst_multiple \
         umount dmsetup blockdev blkid lsblk dd losetup \
         grep cut partprobe find wc fdisk tail mkfs.ext4 mkfs.xfs \
-        dialog cat mountpoint
+        dialog cat mountpoint partx
```

Comment:
If you consider to use blockdev no changes to this file would be needed

Reply:
> I have not used blockdev --rereadpt because on my TW tests it just fails, it complains about a busy device, could it be another issue? 🤷🏽‍♂️

Yes, in fact `blockdev` was my first attempt, however I could not make it work... it complains of a busy device, probably it actually does the re-read operation, but complains of a busy device and returns non zero code. In fact, the `fdisk` call just before is also complaining and outputs a warning message suggesting that partitioning a device in use is bad idea.

Reply:
That fdisk complains is a known issue for years and the reason why we call another tool to re-read the partition table and inform the kernel. ok so this this is good information that blockdev did not work for you. In this case I suggest the other way round and use ```partx -u``` everywhere

Reply:
Yes I am also wondering if we should consider refactoring a little the persistent partition creation to manipulate the disk without having the disk mounted.

So something like:

1. compute if persistent partition is possible and its setup
2. umount anything from the disk
3. repartition
4. mount all the required stuff

I know this is not a straight forward and probably requires a non easy refactor that expands the scope of the issue, but this would get rid of busy state of the block device. This would also get rid of the fdisk warnings and allow `blockdev --rereadpt` operation.

Thoughts on that?

Probably we can do a first attempt with `partx -u` and then consider if such a refactor is worth. Since this is hitting 15.5 code base I think having a minimalistic solution there is also valuable, I doubt we want to apply a refactor there.